src/helpers/tests/format.test.ts

```diff
@@ -17,79 +47,202 @@ describe("getLogLevelCharts", () => {
+      body: "123",
+      mimeType: "image/png",
+      name: "test.png",
+    })
+    const result = await fileToBase64(file)
+    expect(result.substring(0, 14)).toEqual("data:image/png")
+  })
+})
+
+describe(formatDebugRequest, () => {
+  it("should create a request with just text", async () => {
+    const sample = {
+      data: [{ text: "hello world" }],
+      parameters: {},
+    }
+    const expectedResult = JSON.stringify(sample, null, "\t")
+    const result = await formatDebugRequest(
+      sample.data[0].text,
+      null,
+      [],
+      {},
+      {},
+      {}
+    )
+    expect(result).toEqual(expectedResult)
+  })
+
+  it("should create a request with just a file", async () => {
+    const sample = {
+      data: [{ uri: "data:image/png;base64,MTIz" }],
+      parameters: {},
+    }
+    const expectedResult = JSON.stringify(sample, null, "\t")
+    const sampleFiles = createMockFileList([
+      {
+        body: "123",
+        mimeType: "image/png",
+        name: "test.png",
+      },
+    ])
+    const result = await formatDebugRequest("", sampleFiles, [], {}, {}, {})
+    expect(result).toEqual(expectedResult)
+  })
+
+  it("should create a request with text and a file", async () => {
+    const sample = {
+      data: [{ text: "hello world" }, { uri: "data:image/png;base64,MTIz" }],
+      parameters: {},
+    }
+    const expectedResult = JSON.stringify(sample, null, "\t")
+    const sampleFiles = createMockFileList([
+      {
+        body: "123",
+        mimeType: "image/png",
+        name: "test.png",
+      },
+    ])
+    const result = await formatDebugRequest(
+      sample.data[0].text as string,
+      sampleFiles,
+      [],
+      {},
+      {},
+      {}
+    )
+    expect(result).toEqual(expectedResult)
+  })
+
+  it("should create a request with custom parameters", async () => {
+    const sample = {
+      data: [
+        { text: "hello world", modality: "test_2" },
+        { uri: "data:image/png;base64,MTIz", anotherParam: "test_3" },
+      ],
+      parameters: { mode: "test_4" },
+      rootParam: "hello World",
+    }
+    const expectedResult = JSON.stringify(sample, null, "\t")
+    const sampleFiles = createMockFileList([
+      {
+        body: "123",
+        mimeType: "image/png",
+        name: "test.png",
+      },
+    ])
+    const result = await formatDebugRequest(
+      sample.data[0].text as string,
+      sampleFiles,
+      ["test1", "test2", "test3", "test4"],
+      {
+        test1: "root",
+        test2: "textQuery",
+        test3: "file-test.png",
+        test4: "parameters",
+      },
+      {
+        test1: "rootParam",
+        test2: "modality",
+        test3: "anotherParam",
+        test4: "mode",
+      },
+      {
+        test1: "hello World",
+        test2: "test_2",
+        test3: "test_3",
+        test4: "test_4",
+      }
+    )
+    expect(result).toEqual(expectedResult)
+  })
+})
+
+describe(mimeTypeFromDataURI, () => {
```

Comment:
why do you import the actual function instead of writing a string?

Reply:
The function is already imported for testing